src/main/java/com/mycompany/app/Server3333.java

```diff
@@ -12,17 +12,17 @@
     public void bar()
     {
         System.out.println( "Hello World!" );
+       if ("1" == "2") {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.